### PR TITLE
feat(webhooks): allow custom headers on webhook call

### PR DESCRIPTION
**Pull request by amanya**
This will add support for customizing request headers that will be sent to the service called by the webhook stage.

Depends on https://github.com/spinnaker/deck/pull/3589

### CODE REVIEW

File: orca-webhook/src/main/groovy/com/netflix/spinnaker/orca/webhook/WebhookService.groovy [hunk 1]
```diff
@@ -34,9 +35,11 @@ class WebhookService {
   @Autowired
   UserConfiguredUrlRestrictions userConfiguredUrlRestrictions
 
-  ResponseEntity<Object> exchange(HttpMethod httpMethod, String url, Object payload) {
+  ResponseEntity<Object> exchange(HttpMethod httpMethod, String url, Object payload, Object customHeaders) {
     URI validatedUri = userConfiguredUrlRestrictions.validateURI(url)
-    HttpEntity<Object> payloadEntity = new HttpEntity<>(payload)
+    HttpHeaders headers = new HttpHeaders()
+    customHeaders.each{ key, value -> headers.add(key as String, value as String) }
```
Comment (anotherchrisberry) [inline orca-webhook/src/main/groovy/com/netflix/spinnaker/orca/webhook/WebhookService.groovy]: Does this need a guard against a null value, i.e.
```groovy
customHeaders?.each { key, value -> headers.add(key as String, value as String) }
```
?

File: orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy [hunk 2]
```diff
@@ -27,9 +28,7 @@ import spock.lang.Shared
 import spock.lang.Specification
 import spock.lang.Subject
 
-import static org.springframework.test.web.client.match.MockRestRequestMatchers.jsonPath
-import static org.springframework.test.web.client.match.MockRestRequestMatchers.method
-import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.*
```
Comment (anotherchrisberry) [inline orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy]: Prefer not `*`ing imports

File: orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy [hunk 3]
```diff
@@ -51,13 +50,18 @@ class WebhookServiceSpec extends Specification {
...
     when:
+    HttpHeaders headers = new HttpHeaders()
+    headers.add("customHeader", "value")
     def responseEntity = webhookService.exchange(
       HttpMethod.POST,
       "https://localhost/v1/test",
-      ["payload1": "Hello", "payload2": "World!"])
+      ["payload1": "Hello", "payload2": "World!"],
+      ["X-HEADER": "value"]
+    )
```
Comment (anotherchrisberry) [inline orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy]: Should probably also include a test when no headers are supplied.

File: orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy [hunk 4]
```diff
@@ -45,24 +49,43 @@ class WebhookServiceSpec extends Specification {
...
+      customHeaders
+    )
 
     then:
     server.verify()
     responseEntity.statusCode == HttpStatus.OK
     responseEntity.body == ["status": "SUCCESS"]
+
+    where:
+    payload | customHeaders
```
Comment (anotherchrisberry) [inline orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy]: super minor, but you should align the headers on data tables with the data:
```groovy
where:
payload                                     | customHeaders
["payload1": "Hello", "payload2": "World!"] | ["X-HEADER": "value"]
```

**Comment by anotherchrisberry** [comment]
One minor formatting issue and a rebase and we'll get this merged

File: orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy (hunk 4 above)
Comment (amanya) [inline orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy]: Done!

**Comment by anotherchrisberry** [comment]
Sorry, someone merged over you - can you rebase one more time and we'll get this merged?

**Comment by amanya** [comment]
@anotherchrisberry done

**Pull request merged**